### ISSUE TITLE
OsmAnd Edit POI: trim tag names / values before uploading

### ISSUE THREAD

**Issue by laralem** [osmandapp/OsmAnd#11662]
### Description
To my knowledge, OsmAnd doesn't allow moving a POI already in OSM servers. We can move POIs only created by us in OsmAnd and before sending the changes to OSM servers, so how is possible I've moved several existing OSM nodes 0,01m and keeping the same latitude and longitude numbers shown in OSM?

About the tags not added, I've added several "short_name" tags to many existing nodes in OSM. This tag is not in OsmAnd and I have to add it manually. I've made a test right now and OsmAnd changed the name I've changed correctly but didn't added the tag "short_name=AT", see https://www.openstreetmap.org/way/117654285/history This is really a bug or OsmAnd doesn't add tags not known? To others users, it seems I'm making a type of vandalism, since I deleted short names present in "name" tag, but don't add that information to another tag, in this case the "short_name" tag. So this is really annoying and bad.

### How to reproduce?
See for example https://www.openstreetmap.org/node/5198869766/history in JOSM it shows I've changed it's position by 0,01m. In fact, I've checked and found all edits by me in OsmAnd are moving the nodes 0,01m (at least according to JOSM, since I can't see that in OSM website). See for example any node in the changeset https://www.openstreetmap.org/changeset/104600658

About tags not added, just make a test adding "short_name" to any existing OSM POI in a node, send the changeset to OSM and see the result.

### Your Environment
OsmAnd Version: 4.0.0#31957m (released in 2021-05-07) may be a bit more recent than this, after making this edits I've updated OsmAnd.
Android/iOS version: 7.1.1
Device model: Sony Xperia Z5 Compact - Model E5823

**Maps used (online or offline):**   
OsmAnd default offline vector map (don't know the update date since meanwhile I've updated it to the most recent version, just know it's at least from 01-05-2021)

**Comment by vshcherb** [comment]
1. I don't see any moving in the history - https://www.openstreetmap.org/node/3641041050/history
Consistent - 40.2666251, -7.4958084
2. Didn't manage to reproduce (https://www.openstreetmap.org/way/52060712) ,  probably it was a mystical issue...
 https://www.openstreetmap.org/changeset/104949982

**Comment by laralem** [comment]
@vshcherb I still can't add "short_name" tags. Test by your self just adding 1 tag "short_name=XYZ" to anything. I've tried that and OsmAnd made a changeset without changes! How is this possible? At least in JOSM, if you don't change anything you can't upload anything. See this changeset: https://www.openstreetmap.org/changeset/104955818 I just have an explanation, if you analyze that changeset in JOSM, you can see OsmAnd moved the nodes of that area 0,01 m.

Tested in last development version 4.0.0#32017m and OSM live maps.

**Comment by vshcherb** [comment]
Hm... this is strange, I've just tested it myself / added on advanced tab and then submitted via OsmAnd. Do you have another workflow i.e. do you do it somehow via JOSM ?

**Comment by laralem** [comment]
@vshcherb same workflow as you in OsmAnd: select POI > Modify POI > advanced tab > Add tag >  short_name=ES > Upload button. Happened again: https://www.openstreetmap.org/changeset/104965183

I've only used JOSM to check my OsmAnd edits. Of course I also use JOSM to edit OSM data, but not this time.

Since I can't see anything changed in OSM website, I check also in JOSM to see what I've changed, and in it I can only see the nodes of that area changed the position by 0,01m (of course it can be and JOSM error, but that doesn't explain/or could be an explanation not be an error to OSM accepting the change:
Open JOSM > Menu Windows > Changeset Manager > In Changeset ID: just type the ID (in my last example type 104965183) > click download button at right of this field > click "Download changed objects" in context menu while clicking in changeset (or the last button at left right text to the changeset table). Now analyze the "history" of the area (menu Windows > Selection > History button in right panel). Check both tags and nodes history, nothing changed in that area.

In my last edit (https://www.openstreetmap.org/changeset/104965183), I've checked also the nodes history but this time the nodes weren't changed. Anyway, so what was changed then? It's strange. This shouldn't happen and shouldn't be allowed by OSM, I think, I'm not a developer.

I will try a fresh OsmAnd install and see what happens and report it here. Just give me a couple of hours, I've got to make a backup and test different OsmAnd versions.

**Comment by vshcherb** [comment]
Indeed... Please note I'm using offline edits i.e. first edit -> upload. Will try another way as well. Also it's possible to export "*.osc" file of changes (via Edits -> share), probably we could find something inside.

**Comment by laralem** [comment]
@ vshcherb may be I've found part of the problem. At last I know a way to upload "short_name" tags.

I've tested in last Google Play version, OsmAnd 3.9.10, and happened the same https://www.openstreetmap.org/changeset/104969410 but this time I saw something: the tag "short_name" is not in OsmAnd so it's remembered in my cellphone dictionary (Sony software?) and every time I enter the word "sho" I choose the "short_name" shown by the dictionary but like every word in dictionary, it adds an extra space at the end in that tag (useful in normal text). If I enter the value of the tag and select again the tag, the extra space is still there "short_name ". At this point:
* (1) If I leave the "short_name " with the extra space (save the modified POI) and open again the modified POI, the extra space disappeared in "short_name" (at this point I've also tried save and don't save again in OsmAnd, same results) but if I upload this, happens what I've told.
* (2) If I edit the tag again (without leaving the editing dialog) and remove the extra space in "short_name", that works! (https://www.openstreetmap.org/node/3530808737/history my first edit was the workflow (1) and the last edit was this workflow (2) worked.)

I've tested this actions several times. So now we have same and other problems:
1) Why there is no extra space after I save the edit in OsmAnd and open it again? Removes it automatically? If true, great, but in practice it's seems it's like it still there or something since it doesn't upload the new tag.
2) Why OsmAnd uploads an apparently empty changeset?

If needed I can upload my "*.osc" file of changes.
